### DISCUSSION TITLE
b62118fe-681d-44e8-b455-4250b3450b16

### DISCUSSION THREAD

**Discussion by giscus** [varofla/blog_giscus#15] [General]
# b62118fe-681d-44e8-b455-4250b3450b16

ESP32(NimBLE)을 통해 게임패드와 연결해보았습니다.

https://www.varofla.com/b62118fe-681d-44e8-b455-4250b3450b16

**Comment by libqwerty** [comment]
최근에 ESP32를 구매해서 HID를 들여다 보고 있는데, 너무 재미있는 내용이네요 ㅋㅋ
영상이 깨지는 것 같은데, 한 번 체크 해주실 수 있을까용? :pray:

**Reply by varofla** [reply]
재미있으셨다니 다행입니다 :)
영상을 원본 그대로 올리다 보니 용량이 커 로딩이 늦는것 같습니다 ㅜㅜ.. 유튜브를 활용해야겠네요 피드백 감사합니다..!

**Comment by billypark2050** [comment]
ESP32 HID 열심히 파보다가 실패했는데, 재도전하는중에 블로그 내용 보니 도움이 많이 되네요!
결국 Gamepad BLE -> ESP32: BLE=> USB -> PC 이면서 ESP32 에서 재맵핑인거군요,,
ESP32 Gamepad를 오픈소스로만 해보고 프로토콜이니 hid report니 지식의 한계 부딪쳐 포기했었는데 대단하시네요.
글 많이 올려주세요~

**Reply by varofla** [reply]
PC방에서 게임할때 써먹으려 만들고 있었는데 갈 일이 생각보다 없더라고요 ㅋㅋ..
BLE, MQTT 등 배워야할건 늘어나는데 할건 많고.. 분신술을 배워야겠습니다 ㅜㅜ..
